instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -17,6 +17,7 @@ module Instrumentation
             def request(req, body = nil, &block)
               # Do not trace recursive call for starting the connection
               return super(req, body, &block) unless started?
+              return super(req, body, &block) unless OpenTelemetry::Trace.current_span.recording?
```

Comment:
Should this be `unless started? && OpenTelemetry::Trace.current_span.recording?` ?

Reply:
well shoot... I think I got it even more wrong... `unless started? && !OpenTelemetry::Trace.current_span.recording?`

Reply:
Demorgans law, a fried brain, and inverted logic are killing me here:

```ruby
if started? && recording?
  # trace
else 
  # don't trace
end
```

Reply:
Ok, so now this code relies on the fact that there is an active recording span. That means it cannot be used in a standalone instrumentation _without_ wrapping it in  parent recording span. 

Run the example... and there are no exported spans.

```console
@arielvalentin ➜ .../opentelemetry-ruby-contrib/instrumentation/net_http/example (untraced-net-http) $ bundle exec ruby net_http.rb 
I, [2022-06-28T01:43:00.394156 #6728]  INFO -- : Instrumentation: OpenTelemetry::Instrumentation::Net::HTTP was successfully installed with the following options {}
```